### ISSUE TITLE
Enchance example using hyperdht hyperSwarm hyperdrive

### ISSUE THREAD

**Issue by sce9sc** [holepunchto/bare-expo#8]
I am trying to undestand how this example can work using the other hyper libraries and I cannot.  I see that the code that will run bare is a string . How can someone pass other libs such as hyperdrive or hyperswarm ?

I am definitely missing something . Is It possible to have a better example that illustrates how to accomplish this.

Thank you 

PS: I am following this project the last 6 years (before holepunch) . Great job

**Comment by kasperisager** [comment]
To use modules you'll need to bundle your worklet code using https://github.com/holepunchto/bare-pack. This will produce an application bundle whose contents you should pass to `worklet.start()` instead of just a single script. See either https://github.com/holepunchto/bare-ios or https://github.com/holepunchto/bare-android for examples of how this can be done.

Showcasing the various Pear modules is beyond the scope of this template, however. I'd very much like to eventually use a bundle instead of just a script for the worklet example, but I haven't yet found a good way to seamlessly integrate it with the Expo tooling.

**Comment by sce9sc** [comment]
So if I wanted to make the example work with a file called worklet.js I would do as follows.

install bare-pack
```
npm install bare-pack -g
```

create file  ./app/worklet.js and write inside the file 
```
const { IPC } = BareKit

IPC.setEncoding('utf8')
IPC.on('data', (data) => console.log(data))
IPC.write('Hello from Bare!')
```
save it and run 

```
bare-pack --platform ios --arch arm64 --simulator ./app/worklet.js -o ./app/worklet.bundle.js
```

this will produce worklet.bundle.js and in the index.tsx

```

 worklet.start('/app.js', require('./worklet.bundle')).then(() => {
      const { IPC } = worklet

      IPC.setEncoding('utf8')
      IPC.on('data', (data: string) => setReponse(data))
      IPC.write('Hello from React Native!')
    })
  }, [])

```


Is that correct??  But when I run expo run:ios I get

```
 WARN  Route "./worklet.bundle.js" is missing the required default export. Ensure a React component is exported as default. [Component Stack]
 ERROR  ReferenceError: Property 'BareKit' doesn't exist, js engine: hermes [Component Stack]
 WARN  Route "./worklet.js" is missing the required default export. Ensure a React component is exported as default. [Component Stack]
 INFO  
 💡 JavaScript logs will be removed from Metro in React Native 0.77! Please use React Native DevTools as your default tool. Tip: Type j in the terminal to open (requires Google Chrome or Microsoft Edge).
[BareKit] Uncaught SyntaxError: Unexpected token ':'
at Module._extensions..cjs (bare:/worklet.bundle/node_modules/bare-module/index.js:831:32)
at Module._extensions..js (bare:/worklet.bundle/node_modules/bare-module/index.js:816:51)
at Module.load (bare:/worklet.bundle/node_modules/bare-module/index.js:456:38)
at Function.start (bare:/worklet.bundle/shared/worklet.js:129:17)
```

**Comment by kasperisager** [comment]
Close! Your worklet code must _not_ reside in the `app/` directory, however, as Metro will otherwise interpret it as React components. Make a dedicated directory for your worklet code instead.

**Comment by sce9sc** [comment]
Ok I just created a src dir and put both worklet.js and worklet.bundle.js inside 

changed index.js to 

``` 
 worklet.start('/app.js', require('../src/worklet.bundle')).then(() => {
      const { IPC } = worklet

      IPC.setEncoding('utf8')
      IPC.on('data', (data: string) => setReponse(data))
      IPC.write('Hello from React Native!')
    })

```

and run expo again . I got 

```
Uncaught SyntaxError: Unexpected token ':'
at Module._extensions..cjs (bare:/worklet.bundle/node_modules/bare-module/index.js:831:32)
at Module._extensions..js (bare:/worklet.bundle/node_modules/bare-module/index.js:816:51)
at Module.load (bare:/worklet.bundle/node_modules/bare-module/index.js:456:38)
at Function.start (bare:/worklet.bundle/shared/worklet.js:129:17)
```

**Comment by kasperisager** [comment]
What does `worklet.bundle.js` contain? From the syntax error it sounds like it's a raw bundle, but it should be wrapped.

**Comment by sce9sc** [comment]
worklet.bundle.js

```
module.exports = "352\n{\"version\":0,\"id\":\"dd3c6070bcd2c8f5d934c564f4916aacbe35e33f048bfb8fae1093179af5c396\",\"main\":\"/app/worklet.js\",\"imports\":{},\"resolutions\":{\"/app/worklet.js\":{\"#package\":\"/package.json\"},\"/package.json\":{}},\"addons\":[],\"assets\":[],\"files\":{\"/app/worklet.js\":{\"offset\":0,\"length\":170,\"mode\":420},\"/package.json\":{\"offset\":170,\"length\":1221,\"mode\":420}}}\nconst { IPC } = BareKit\n\nIPC.setEncoding('utf8')\nIPC.on('data', (data) => console.log(data))\nIPC.write('Hello from Bare!')\n\nconst main = ()=>{}\n\n// module.exports = main;{\n  \"private\": true,\n  \"name\": \"bare-expo\",\n  \"description\": \"Example of embedding Bare in an Expo application\",\n  \"main\": \"expo-router/entry\",\n  \"scripts\": {\n    \"start\": \"expo start\",\n    \"android\": \"expo run:android\",\n    \"ios\": \"expo run:ios\",\n    \"build:worklet\":\"bare-dev bundle -o app/worklet.bundle.js app/worklet.js\"\n  },\n  \"repository\": {\n    \"type\": \"git\",\n    \"url\": \"git+https://github.com/holepunchto/bare-expo.git\"\n  },\n  \"author\": \"Holepunch\",\n  \"license\": \"Apache-2.0\",\n  \"bugs\": {\n    \"url\": \"https://github.com/holepunchto/bare-expo/issues\"\n  },\n  \"homepage\": \"https://github.com/holepunchto/bare-expo#readme\",\n  \"dependencies\": {\n    \"events\": \"^3.3.0\",\n    \"expo\": \"^52.0.17\",\n    \"expo-build-properties\": \"^0.13.1\",\n    \"expo-constants\": \"^17.0.3\",\n    \"expo-linking\": \"^7.0.3\",\n    \"expo-router\": \"^4.0.11\",\n    \"expo-system-ui\": \"^4.0.5\",\n    \"react\": \"^18.3.1\",\n    \"react-dom\": \"^18.3.1\",\n    \"react-native\": \"^0.76.3\",\n    \"react-native-bare-kit\": \"^0.4.0\",\n    \"react-native-safe-area-context\": \"^4.14.0\"\n  },\n  \"devDependencies\": {\n    \"@types/react\": \"^18.3.1\",\n    \"bare-dev\": \"^0.14.11\",\n    \"prettier\": \"^3.3.3\",\n    \"prettier-config-standard\": \"^7.0.0\",\n    \"typescript\": \"^5.3.3\"\n  }\n}\n"

```

**Comment by kasperisager** [comment]
Ah, I just spotted the issue; your bundle is of course a bundle and so the filename should reflect that as Bare will otherwise treat it as a JavaScript module, hence the syntax error:

```js
worklet.start('/app.bundle', require('../src/worklet.bundle'))
```

**Comment by sce9sc** [comment]
Thank you very much . it works now . 

Now it is documented :) . I will try to create an example having hyperswarm as well and also other libs for reference.

**Comment by sce9sc** [comment]
One last question . 

In order to add fs and other modules what do I need to do ?

Do I need to install all the bare modules like 
```
npm install bare-fs

```

and then go to package.json and add "fs": "npm:bare-node-fs" to the dependencies

```
 "dependencies": {
   "bare-fs": "^4.0.1",
   "fs": "npm:bare-node-fs",
   ...
}

```
 
run 'npm install' again


Edit the worklet.js file 
```
const { IPC } = BareKit
const fs = require('fs');

IPC.setEncoding('utf8')
IPC.on('data', (data) => console.log(data))
IPC.write('Hello from Bare!')

```

Run again bare-pack

```
bare-pack --platform ios --arch arm64 --simulator ./app/worklet.js -o ./app/worklet.bundle.js
```

and expo 


What I get is 

```
[BareKit] Uncaught SyntaxError: Unexpected token 'o', "or (stream"... is not valid JSON
at JSON.parse (<anonymous>)
at Module._extensions..json (bare:/worklet.bundle/node_modules/bare-module/index.js:875:28)
at Module.load (bare:/worklet.bundle/node_modules/bare-module/index.js:456:38)
at Module._extensions..js (bare:/worklet.bundle/node_modules/bare-module/index.js:803:18)
at Module.load (bare:/worklet.bundle/node_modules/bare-module/index.js:456:38)
at Module._extensions..bundle (bare:/worklet.bundle/node_modules/bare-module/index.js:930:28)
at Module.load (bare:/worklet.bundle/node_modules/bare-module/index.js:456:38)
at Function.start (bare:/worklet.bundle/shared/worklet.js:129:17)
```

**Comment by kasperisager** [comment]
There's no need to map `fs` to `bare-fs` prior to importing it, you can just do:

```
const fs = require('bare-fs')
```

For the other issue, see https://github.com/holepunchto/bare-pack#linking.

**Comment by sce9sc** [comment]
Why don't I need to map ? What about if I want to use other modules that use fs . 

Thank you, it would have taken ages to find it ( if at all )

I used --linked flag and now the bare-pack command is as shown below. 

```
bare-pack --linked  --platform ios --arch arm64 --simulator ./src/worklet.js -o ./src/worklet.bundle.js
```

I also used "Consuming dependencies using Import Maps"  found  [here](https://docs.pears.com/bare-reference/nodejs-compatibility-with-bare#consuming-dependencies-using-import-maps)

From what I undestood I need to run bare-pack for each platform and arch. Is this correct ?

I really appreciate the help that you are giving me.
